src/python/zquantum/core/wavefunction.py

```diff
@@ -46,23 +46,26 @@ def __init__(
             )
 
         try:
-            self._wavefunction = np.asarray(amplitude_vector, dtype=complex)
+            self._amplitude_vector = np.asarray(amplitude_vector, dtype=complex)
         except TypeError:
-            self._wavefunction = Matrix(amplitude_vector)
+            self._amplitude_vector = Matrix(amplitude_vector)
 
-        self._check_sanity(self._wavefunction)
+        self._check_sanity(self._amplitude_vector)
 
     @property
     def amplitudes(self):
-        return _cast_sympy_matrix_to_numpy(self._wavefunction, complex=True)
+        if _free_symbols(self._amplitude_vector):
```

Comment:
I would change `_free_symbols` to `_check_free_symbols` or `_has_free_symbols`.

Reply:
This function  returns free symbols, i.e. it is not a boolean function.